docs/profiler.md

```diff
@@ -1,136 +1,196 @@
 ```
 
-This formula gives the probability of finding a valid proof in one data pass. The node will perform multiple passes if needed.
+The aforementioned formula gives the probability (with an output of 1 being 100%) of finding a valid PoST in one data pass. The node will perform multiple passes if PoST is not found in one data pass.
+
+Please note that PoST generation speed scales *linearly* with the number of CPU cores used (assuming that they are equally fast) and *inversely* with the number of nonces in groups of 16 (32 nonces should be twice as slow as 16). This effect might not manifest itself until a high number of nonces is used. The reason for this is that in most setups, the hard disk drive (HDD) speed will become the limiting factor (as a spinning disk drive is used in most computers) if a low number of nonces is used.
+
+So, if we increase the CPU cores, the faster a valid PoST will be generated. And if you increase the number of nonces, the slower a valid PoST will be generated due to the increased usage of the HDD which results in the HDD speed being the limiting factor. Knowing this will help you make a better assessment of the profiler tool flags and its output.
 
-Please note that proving speed linearly scales with the number of cores used (assuming that they're equally fast), but inversely scales with the number of nonces in groups of 16 (32 nonces should be twice as slow as 16). This effect might not manifest itself until a high number of nonces is used. The reason is that in most setups the hard disk speed will be the limiting factor if a low number of nonces is used.
+## How to interpret the profiler output
 
-## How to interpret the results
+We will now understand how to interpret the profiler output. Let us run the following command (for reference, all commands are being run on a 2020 MacBook Air M1 16GB/256GB):
 
 ```
-./profiler --data-size 1 --threads=1 --data-file data.bin --nonces=64
+./profiler --threads=1
+```
+
+In the above terminal command, we are running the profiler in `proving` mode and have provided the following options:
+
+* The number of threads to be used is 1 (instead of the default 4).
+* For all other options, default values (e.g., 64 for nonces) are to be used ([see above](#understanding-the-options-and-commands)).
+
+After running this command, we get the following output:
+
+```bash
 {
-  "time_s": 12.09140029,
-  "speed_gib_s": 0.41351703525481415
+  "time_s": 10.316020166,
+  "speed_gib_s": 0.19387321542775668
 }
 ```
 
-Here we see the following
-* 1GiB file `data.bin` was used for the benchmark
-* 1 thread was used
-* 64 nonces were used
+The output means that it took 10.31 seconds to find a valid PoST and the speed was 0.19 GiB/s.
+From the [formula above](#nonce-estimation-formula), we know that the probability of finding a proof with 64 nonces is 79.39%. Therefore, there is a ~20% chance that at least two passes are necessary (and a ~ 0.20^x chance that more than x passes are necessary).
```

Comment:
>The output means that it took 10.31 seconds to find a valid PoST and the speed was 0.19 GiB/s.

The benchmark ran for 10.31 seconds, not looking for a PoST.  Wording it like this gives a false impression that looking for a PoST in a 1GiB file takes 10s, which is clearly not true - given the speed of 0.19GiB/s, it would take roughly 5s (if a single pass is needed).

The profiler runs multiple passes over the file, up to the configured `--duration` - the longer the duration the more accurate the result is (it averages).

Perhaps it's best to just remove `time_s` from the output to avoid confusion. Wdyt?

Reply:
I see. So the profiler only calculates the speed, in GiB/s, of performing passes over PoS data of the given size? The output speed is then how fast can our node go over PoS data of given size when generating the PoST. The longer the profiler runs for, the more accurate the speed will be since in real world scenario, generally, the node takes more time and multiple passes to come up with the proof. Correct? 

And I agree on removing `time_s`. Then, for the output, perhaps we can print a string saying that "Your node can perform the PoST generation process at a estimated speed of X GiB/s".

Also, I think we should refactor the profiler to bench in SUs (atleast 1 SU) as PoS data size will always be a multiple of 64.

Reply:
>So the profiler only calculates the speed, in GiB/s, of performing passes over PoS data of the given size?

The speed is not a factor of the size of the data, it remains the same no matter the size, otherwise correct.

>And I agree on removing time_s. Then, for the output, perhaps we can print a string saying that "Your node can perform the PoST generation process at a estimated speed of X GiB/s".

It prints JSON because it was meant to be processed automatically by SMAPP. We could have a flag for to control human/machine output format.